airflow/executors/base_executor.py

```diff
@@ -284,8 +289,11 @@ def trigger_tasks(self, open_slots: int) -> None:
                     self.log.info("queued but still running; attempt=%s task=%s", attempt.total_tries, key)
                     continue
                 # Otherwise, we give up and remove the task from the queue.
-                self.log.error(
-                    "could not queue task %s (still running after %d attempts)", key, attempt.total_tries
+                self.task_context_logger.error(
+                    "could not queue task %s (still running after %d attempts)",
```

Comment:
I think it's confusing for a user to see that a task couldn't be queued because it's currently running. Is there any way to make this log message more useful/intuitive?

Reply:
Do you have any suggestions?

Reply:
No, but I've run into this when troubleshooting and it always confuses me 😅

Reply:
You're the perfect candidate to create a meaningful error message then :)

Reply:
I updated the error message, please let me know if that looks better to you

Reply:
Definitely better. Do you think it makes sense to clarify that "running" here is the executor state rather than the Airflow state?

Reply:
Would "still running from executor point of view after %d attempts" be better?

Reply:
What about...

> Could not queue task %s. Learn more: https://airflow.apache.org/docs/apache-airflow/stable/troubleshooting.html#still-running-on-executor

And add a blurb in [troubleshooting.rst](https://github.com/apache/airflow/blob/main/docs/apache-airflow/troubleshooting.rst) explaining what the comment says (triggerer race condition or the task is soon going to be marked failed)?

Reply:
That works too. Thanks for the proposal!

Reply:
Sorry for the late response, I was thinking about this a bit and it looks like you've settled on something, but an alternative just came to me... take it or leave it, of course... but I'll suggest an option:

```Failed to queue task %s after %d attempts; executor reports task as currently running. ``` 

perhaps with the executor name/id in there, perhaps with the attempt count in there, maybe sub "already" instead of "currently"....

Reply:
I like this proposal. @RNHTTR, what do you think? TBH @RNHTTR, I have nothing against writing a section in [troubleshooting.rst](https://github.com/apache/airflow/blob/main/docs/apache-airflow/troubleshooting.rst) about this special use case but since I have never encountered this use case I dont feel strongly confident describing it (since I am not sure how it happens). Or maybe you could describe it @RNHTTR? That'd be helpful

Reply:
> I dont feel strongly confident describing it (since I am not sure how it happens)

This is my problem too, which is why I'm hoping we can come up with something more useful or just not surface this log. Usually when I encounter this, I mostly ignore it and look for something else that's meaningful.

In my opinion, `executor reports task is currently running` is still tricky to users. If I understand correctly, it's reporting the state of the task in, for example, celery (that is, `running`), which is different than the Airflow state of the task (`queued`). I think surfacing this log as written will only confuse users more than if it didn't show up at all.

Reply:
I think the fact that we are trying few times in order to avoid race condition is something that should be in the info:

```
self.log.info("The task is queued but still running. Trying for %s time to avoid deferred task race condition. Task=%s ", attempt.total_tries, key)
```

This is pure information, that might help with diagnostics. No action from user is expected. 


But then when we reach the time to avoid the race conditions and we are reasonably sure this is externally killed task we should do this (when you look at `can_try_again()` it allows you to run multiple (any number of) attempts unless 10 seconds pass:

```
   self.task_context_logger.error(
                    "Could not queue task %s as it is seen as still running after %d attempts (tried for 10 seconds). It looks like it was killed externally. Look for extermal reasons why it has been killed (likely a bug or deployment issue).",
```

Likely the 10 seconds should be somewhat extracted as a constant to pass to can_try_again and used in the message rather than hardcoding the 10 seconds.

@malthe  @dstandish - I think you were implementing the changes to handle those cases - is my understanding and proposal reasonable (if you still remember anything about it, that is :) )

Reply:
I updated the text to follow this proposal. I'll update it if something better comes up

Reply:
I think the most context I can provide at this point (having forgot most of the details) is the description in the pull request: https://github.com/apache/airflow/pull/21316.

Reply:
I think this is still really confusing. 

> Could not queue task %s as it is seen as still running after %d attempts (tried for %d seconds). It looks like it was killed externally. Look for external reasons why it has been killed (likely a bug or deployment issue).

* What could not queue the task?
* Seen as still running by what?
* What does externally mean?
* Why is this likely a bug or deployment issue?

I had a convo with @wolfier @collinmcnulty @seanmuth and others, and below is what we landed on. IMO it's too long to share as a log message and should be added to [troubleshooting.rst](https://github.com/apache/airflow/blob/main/docs/apache-airflow/troubleshooting.rst):

> **State mismatch between Airflow Metadata Database and Executor**
> This indicates that when the scheduler queried the Airflow database, it observed that the task instance's status was `queued`, but the status according to the executor was `running`. This mismatch persisted for several attempts. As a result, Airflow will not attempt to queue the task. It's possible that something has gone wrong in the executor, and the task needs to be cleared.

Reply:
alright here's a proposal: don't change the message for this PR.  in this PR, just add the "send to task" part. in followup, someone can improve the language.  what do yall think?

Reply:
Love it

Reply:
Done

Reply:
The PR is now merged. @RNHTTR, if you want to modify the error message, you should be able to do so on top of this change